gobblin-metastore/src/main/java/gobblin/metastore/FsStateStore.java

```diff
@@ -174,11 +172,13 @@ public void putAll(String storeName, String tableName, Collection<T> states)
     } finally {
       closer.close();
     }
+
+    Path tablePath = new Path(new Path(this.storeRootDir, storeName), tableName);
+    HadoopUtils.renamePath(this.fs, tmpTablePath, tablePath);
```

Comment:
@kadaan These files are temporary and only contain metadata. They are deleted after the Gobblin run. These files should not be placed in S3, instead they should go in the HDFS file system (even if it is ephemeral).